crates/agglayer-types/src/lib.rs

```diff
@@ -1,28 +1,58 @@
 use std::collections::{BTreeMap, BTreeSet};
 
-use reth_primitives::{Address, Signature, U256};
-use serde::{Deserialize, Serialize};
-
-use crate::bridge_exit::NetworkId;
-use crate::keccak::keccak256_combine;
-use crate::local_exit_tree::hasher::Keccak256Hasher;
-use crate::local_state::{LocalNetworkStateData, StateCommitment};
-use crate::{
-    bridge_exit::BridgeExit,
+pub use pessimistic_proof::local_exit_tree::hasher::Keccak256Hasher;
+use pessimistic_proof::{
+    bridge_exit::{BridgeExit, TokenInfo},
     imported_bridge_exit::ImportedBridgeExit,
-    keccak::Digest,
+    keccak::{keccak256_combine, Digest},
+    local_balance_tree::LocalBalancePath,
+    local_state::{LocalNetworkStateData, StateCommitment},
     multi_batch_header::MultiBatchHeader,
     nullifier_tree::{NullifierKey, NullifierPath},
 };
-use crate::{bridge_exit::TokenInfo, local_balance_tree::LocalBalancePath};
+use reth_primitives::{Address, Signature, U256};
+use serde::{Deserialize, Serialize};
 
 #[derive(Debug, thiserror::Error)]
 pub enum Error {
-    #[error("Conversion error: {0}")]
+    #[error("Converstion error: {0}")]
     TypeConversion(&'static str),
 }
 
+pub type EpochNumber = u64;
+pub type CertificateIndex = u64;
+pub type CertificateId = [u8; 32];
+pub type Hash = [u8; 32];
 pub type Height = u64;
+pub use pessimistic_proof::bridge_exit::NetworkId;
+
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub struct CertificateHeader {
+    pub certificate_id: CertificateId,
+    pub network_id: NetworkId,
+    pub height: Height,
+    pub epoch_number: Option<EpochNumber>,
+    pub certificate_index: Option<CertificateIndex>,
+    pub local_exit_root: Hash,
```

Comment:
```suggestion
    pub new_local_exit_root: Hash,
```

Given that one Certificate provides the prev and new